### DISCUSSION TITLE
Meow equivalent to Vim's `dw`

### DISCUSSION THREAD

**Discussion by asymmetric** [meow-edit/meow#535] [Q&A] [answered]
In the tutorial, there is this:

```
---> This sentence pencil has vacuum extra words in it.
       This sentence has vacuum words in it.
```

If I follow the tutorial and:

1. place the cursor at the beginning of `pencil` (on the `p`)
2. type `es`

What I get is something like: `This sentence  has vacuum extra words in it.` -- note the double space between "sentence" and "has".

In other words:

1. `es` acts like Vim's `diw`
2. the tutorial is not mentioning that to get the first line to look like the second, one also needs to use `d` to delete individual characters

Am I doing something wrong, or is this a problem in the tutorial?

Moreover, is there an idiomatic way to also delete the space after a word? This seems like the most common behavior, at least for me.
Is this related to https://github.com/meow-edit/meow/discussions/87, i.e. the cursor always being between two characters?

![meow](https://github.com/meow-edit/meow/assets/101816/5d7aa65e-2c05-4d89-ba5e-35137c7a5dda)

It looks like the `e` has selected the space after the word, and I would expect `s` to delete it, but actually the cursor is between the "l" and the space?

**Comment by eshrh** [comment]
You're right, the tutorial is definitely missing something there.

I'm not 100% sure what the tutorial is intending to make you do there, but I suspect it's the way I do it: put your cursor in-between the the works "sentence" and "pencil", then call "e" which selects the first space and the next word, and then delete. A block cursor makes what's going on slightly unclear.

I don't think there's an idiomatic way to automatically delete the space - which space would we delete?

**Reply by asymmetric** [reply]
> I don't think there's an idiomatic way to automatically delete the space - which space would we delete?

@eshrh in vim, `w` moves by a word, which

```
consists of a sequence of letters, digits and underscores, or a sequence of other non-blank characters, separated with white space (spaces, tabs, <EOL>).
```

So the idea here would be to delete the space(s) after the characters. This seems to me to be the most intuitive behaviour.

**Comment by DogLooksGood** [comment, marked as answer]
Use `bounds-of-thing e(symbol)` to select the symbol and one space.
